server/resources/resource_management.go

```diff
@@ -593,20 +593,19 @@ func CreateResource(
 		return err
 	}
 
-	if resourceData, ok := context["resource"].(map[string]interface{}); ok {
-		dataMap = resourceData
-	}
-
 	//Validation
-
 	if _, ok := context[goValidationContextKey]; ok {
-		err = resourceSchema.ValidateGoOnCreate(dataMap)
-		if err != nil {
+		if err := resourceSchema.ValidateOnCreate(dataMap); err != nil {
```

Comment:
as discussed offline: UPDATE does the same thing [differently](https://github.com/cloudwan/gohan/blob/master/server/resources/resource_management.go#L807). as a result, fields set by PreUpdate handlers may be lost.

I think we should unify the way ValidateOnCreate and ValidateOnUpdate are called in this file.